### PR TITLE
feat: update dependencies inside the component model

### PR DESCRIPTION
**Pull request by zkochan**
todo:
- [x] test that the same dep is not shown twice when it is both in the model and in the root policy

## Proposed Changes

-
-
-

### CODE REVIEW

File: scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts [hunk 1]
```diff
@@ -1155,19 +1155,38 @@ export class DependencyResolverMain {
...
   }: {
     rootDir: string;
     variantPoliciesByPatterns: Record<string, VariantPolicyConfigObject>;
     componentPoliciesById: Record<string, any>;
+    components: Component[];
   }): Promise<OutdatedPkg[]> {
+    const componentModelVersions: ComponentModelVersion[] = [];
+    await Promise.all(
+      components.map(async (component) => {
+        const depList = await this.getDependencies(component);
```
Comment (GiladShoham) [inline scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts:1172]: I wonder if we need to filter our extension deps here.
as putting them in the root policy won't change then anyway (and they shouldn't be there)

Comment (GiladShoham) [reply scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts:1172]: Also wonder what will happen with peer deps

File: scopes/workspace/workspace/workspace.ts [hunk 2]
```diff
@@ -1886,10 +1886,13 @@ needed-for: ${neededFor || '<unknown>'}`);
     const { componentConfigFiles, componentPoliciesById } = await this._getComponentsWithDependencyPolicies();
     const variantPatterns = this.variants.raw();
     const variantPoliciesByPatterns = this._variantPatternsToDepPolicesDict(variantPatterns);
+    const compDirMap = await this.getComponentsDirectory([]);
+    const components = compDirMap.toArray().map(([component]) => component);
```
Comment (GiladShoham) [inline scopes/workspace/workspace/workspace.ts:1890]: why don't you just use `this.list()` ?

File: scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts (hunk 1 above)
Comment (zkochan) [reply scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts:1172]: Regarding peer dependencies.

I have created a component with a peer dependency:
image

but when I import it, it shows the peer as a runtime dependency:
image

Comment (zkochan) [reply scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts:1172]: extensions and peers in the model are skipped by the update command.

File: scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts [hunk 3]
```diff
@@ -1155,19 +1155,43 @@ export class DependencyResolverMain {
...
+        depList.forEach((dep) => {
```
Comment (GiladShoham) [inline scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts:1173]: @zkochan why not use depList.filter?

Comment (zkochan) [reply scopes/dependencies/dependency-resolver/dependency-resolver.main.runtime.ts]: changed to filter

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**